k4Reco/Overlay/components/OverlayTiming.cpp

```diff
@@ -0,0 +1,344 @@
+          if (this_stop <= (BX_number_in_train - _BX_phys) * _T_diff) {
+            info() << "Skipping collection " << name << " as it is not in the integration window" << endmsg;
+            continue;
+          }
+
+          auto ocoll = edm4hep::SimCalorimeterHitCollection();
+          auto mapp  = cellIDsMap[name];
+          for (auto elem : backgroundEvent.get<edm4hep::SimCalorimeterHitCollection>(name)) {
+            if (mapp.find(elem.getCellID()) == mapp.end()) {
+              // There is no Hit at this position -- the new hit can be added, if it is not outside the window
+              auto calhit = edm4hep::MutableSimCalorimeterHit();
+              bool add    = false;
+              for (auto& contrib : elem.getContributions()) {
+                if ((contrib.getTime() + timeOffset > this_start) && (contrib.getTime() + timeOffset < this_stop)) {
+                  add = true;
+                  // TODO: Make sure a contribution is not added twice
+                  auto newcontrib = contrib.clone();
+                  calhit.addToContributions(newcontrib);
+                  ocaloHitContribs.push_back(newcontrib);
+                }
+              }
+              if (add) {
+                calhit.setCellID(elem.getCellID());
+                calhit.setEnergy(elem.getEnergy());
+                calhit.setPosition(elem.getPosition());
+                mapp[calhit.getCellID()] = calhit;
+                info() << "Adding new hit with cellID: " << elem.getCellID() << endmsg;
+                info() << calhit.getContributions().size() << endmsg;
+              }
+            } else {
+              // there is already a hit at this position....
+              // TODO: get the hit
+              auto& calhit = mapp[elem.getCellID()];
+              for (auto& contrib : elem.getContributions()) {
+                if ((contrib.getTime() + timeOffset > this_start) && (contrib.getTime() + timeOffset < this_stop)) {
+                  // TODO: Make sure a contribution is not added twice
+                  auto newcontrib = contrib.clone();
+                  calhit.addToContributions(newcontrib);
+                  ocaloHitContribs.push_back(newcontrib);
+                  // info() << "Hit found, adding contribution with cellID: " << elem.getCellID() << endmsg;
+                  // info() << calhit.getContributions().size() << endmsg;
+                }
+              }
+            }
+          }
+          for (auto& [cellID, hit] : mapp) {
+            // why clone?
+            info() << "CellID: " << cellID << ", contributions: " << hit.getContributions().size() << endmsg;
+            // info() << "begin: " << hit.getContributions().begin() << endmsg;
+            // info() << "end: " << hit.getContributions().end() << endmsg;
+            ocoll->push_back(hit.clone());
+          }
+          osimCaloHits["New" + name] = std::move(ocoll);
```

Comment:
```suggestion
          osimCaloHits["New" + name] = std::move(ocoll);
```
Make the "New" name configurable.

Reply:
"New" is not used anymore, names for the output collections are given